pkg/config/config.go

```diff
@@ -164,11 +164,17 @@ func (gc *Config) KappDeployRawOptions() []string {
 	gc.dataLock.RLock()
 	defer gc.dataLock.RUnlock()
 
+	kappOptions := make([]string, 0)
+
 	// Configure kapp to keep only 5 app changes as it seems that
 	// larger number of ConfigMaps negative affects other controllers on the cluster.
 	// Eventually kapp can be smart enough to keep minimal number of app changes.
 	// Set default first so that it can be overridden by user provided options.
-	return append([]string{"--app-changes-max-to-keep=5"}, gc.data.kappDeployRawOptions...)
+	// return append([]string{"--app-changes-max-to-keep=5"}, gc.data.kappDeployRawOptions...)
+	kappOptions = append(kappOptions, "--app-changes-max-to-keep=5")
+	kappOptions = append(kappOptions, "--apply-timeout=5m")
```

Comment:
I think you are planning to change the `wait-timeout` and not `apply-timeout`?

Reply:
The idea of changing apply-timeout is to avoid scenarios where we wait for a resource with the exists annotation for a very long time (_exists_ is an operation so the waiting for the resource to exists is accounted for in the apply-timeout)